src/doc/versions.txt

```diff
@@ -0,0 +1,20 @@
+# This file is used by the version selector of the sage doc
```

Comment:
How about putting this file in one of the `_static` directories instead?
Then `fetchVersions` could get it from one of the deployed documentation sites, instead of having to refer to raw.githubusercontent.com

Reply:
That is how I did it first, but discarded it thinking that it is a mistake.

The selector is to list currently supported docs (of the stable sage versions that released around in the last two years). The list of supported (available) docs should be independent of the deployed doc site. If it were, the site deploying an old doc would list only older docs in the selector, some of which may not be available anymore.

Using raw.githubusercontent.com is how to list only supported docs in the selector even in sites deploying an old doc. A realistic scenario is when a user is viewing a locally built doc for an old sage version.

Reply:
I mean, it could refer to wherever we deploy the latest doc, no?

Reply:
Ah, I misunderstood you. You mean to fetch the file from a prominent deployed doc site like (1) "doc-release--sagemath.netlify.app" or (2) "doc.sagemath.org"? 

That would work. Then I would put it to the root of the doc, rather than "_static". I prefer (1).

Reply:
> I mean, it could refer to wherever we deploy the latest doc, no?

Yes.

Reply:
(1) will reflect the change of the list faster than (2).

Reply:
Sounds good to me.

Reply:
There is a problem in fetching "versions.txt" from "doc-release--sagemath.netlify.app" when the browser is showing the doc from file system, because of CORS policy. The netlify server does allow that. Hence the version selector does not work in this case.

There is no such problem with "raw.githubusercontent.com", which seems permissive.

Why is it that we should avoid "https://raw.githubusercontent.com/sagemath/sage/develop/..."?  I agree that "doc-release--sagemath.netlify.app" is explicitly related with a "release".

Reply:
It is not only from the file system...

![Screen Shot 2024-07-01 at 4 00 12 PM](https://github.com/sagemath/sage/assets/1456101/168c4a04-9c6b-4e72-820c-379e3a598063)

Reply:
Adding `_headers` file with content
```
/html/en/versions.txt
  Access-Control-Allow-Origin: *
  Access-Control-Allow-Methods: GET
  Access-Control-Allow-Headers: Content-Type
```
seems to make the doc site more permissive related with CORS. We may add the `_headers` file when we deploy "doc-release--sagemath.netlify.app" . 

But this is getting complicated... I need an excuse why I should do this :-)

Reply:
I did it anyway. See the PR description.